src/lib.rs

```diff
@@ -240,3 +242,59 @@ pub struct SigElement {
     pub start: usize,
     pub end: usize,
 }
+
+#[derive(Debug, Clone, RustcDecodable, RustcEncodable)]
+pub struct Borrows {
+    pub ref_id: Id,
+    pub assignments: Vec<Assignment>,
```

Comment:
Ah so, you compute the 'scope span' on the fly and that depends on some assignments and some other data for a variable (loans?). So either we compute this in the compiler and pass the span explicitly, but don't need to pass assignments and moves, or we pass a load of data and compute in the tool, which gives a faster save-analysis? Do you have an idea for how long it will take to do this in the compiler? I would prefer to do the compiler option and plan to do this on demand in the near future, rather than do this in the RLS.